### PR TITLE
Change all references to sdkerrors to new errorsmod and ibcerrors

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #2896

This PR changes all occurences of `sdkerrors.Wrap/Wrapf` to `errorsmod.Wrap/Wrapf` and all refersnces to default sdk errors to use `ibcerrors` which were added in #3184.

### Commit Message / Changelog Entry

```bash
chore: Migrated all sdk errors to the new errors go module.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/types/transfer_authorization.go [hunk 1]
```diff
@@ -29,18 +30,18 @@ func (a TransferAuthorization) MsgTypeURL() string {
...
 		if allocation.SourceChannel == msgTransfer.SourceChannel && allocation.SourcePort == msgTransfer.SourcePort {
 			limitLeft, isNegative := allocation.SpendLimit.SafeSub(msgTransfer.Token)
 			if isNegative {
-				return authz.AcceptResponse{}, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, "requested amount is more than spend limit")
+				return authz.AcceptResponse{}, errorsmod.Wrapf(ibcerrors.ErrInsufficientFunds, "requested amount is more than spend limit")
 			}
 
 			if !isAllowedAddress(ctx, msgTransfer.Receiver, allocation.AllowList) {
-				return authz.AcceptResponse{}, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "not allowed address for transfer")
+				return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidAddress, "not allowed address for transfer")
```
Comment (chatton) [inline modules/apps/transfer/types/transfer_authorization.go:44]: note: Wrapf was changed to Wrap as there was one argument being passed

File: modules/apps/transfer/types/transfer_authorization.go [hunk 2]
```diff
@@ -64,44 +65,44 @@ func (a TransferAuthorization) Accept(ctx sdk.Context, msg sdk.Msg) (authz.Accep
...
 		if err := host.ChannelIdentifierValidator(allocation.SourceChannel); err != nil {
-			return sdkerrors.Wrap(err, "invalid source channel ID")
+			return errorsmod.Wrap(err, "invalid source channel ID")
 		}
 
 		found := make(map[string]bool, 0)
 		for i := 0; i < len(allocation.AllowList); i++ {
 			if found[allocation.AllowList[i]] {
-				return sdkerrors.Wrapf(ErrInvalidAuthorization, "duplicate entry in allow list %s")
+				return errorsmod.Wrapf(ErrInvalidAuthorization, "duplicate entry in allow list %s", allocation.AllowList[i])
```
Comment (chatton) [inline modules/apps/transfer/types/transfer_authorization.go:105]: Wrapf was being used but no arg was being passed. I've now passed the duplicate entry.

**Review by chatton — COMMENTED** [review]
There were two linting errors that were also resolved here, I highlighted them as they are difficult to pick out in a diff like this.

**Review by damiannolan — APPROVED** [review]
Looks good to me! 🙏

**Review by colin-axner — APPROVED** [review]
Fabulous work @chatton :racehorse:

**Pull request merged**